### PR TITLE
Add custom.css resource after diazo bundle

### PR DESCRIPTION
**Pull request by MrTango**
Insert virtual custom.css bundle after Diazo bundle.

### CODE REVIEW

**Comment by mauritsvanrees** [comment]
For me it is fine to include this in 5.2. It makes small site-specific styling changes easier, which is good.
Does the @plone/framework-team want to weigh in on this?

The upgrade step would need to be backported to 5.2 then.

**Comment by ale-rt** [comment]
I added a note in the "FWT Meeting Protocol 2020-05-05" shared document!

**Comment by mauritsvanrees** [comment]
I see the team meeting was supposed to be yesterday. Was this discussed?

(Needs an upgrade step in plone.app.upgrade for 5.2, like was added for 6.0 already.)

**Comment by ale-rt** [comment]
We rescheduled the meeting to next Tuesday (2020-05-12)!

File: Products/CMFPlone/resources/browser/styles.py [hunk 1]
```diff
@@ -143,6 +152,18 @@ def styles(self):
...
+        custom_css = {
+            'rel': 'stylesheet',
+            'conditionalcomment': '',
+            'src': "{0}/custom.css?timestamp={1}".format(
+                self.site_url,
+                self.custom_css_timestamp,
+            ),
+            'bundle': 'custom-css'
+        }
+        result.append(custom_css)
```
Comment (ale-rt) [inline Products/CMFPlone/resources/browser/styles.py:166]: Maybe it is better to add some condition to not always append the custom css?
It makes no sense if it is empty, right?

**Comment by ale-rt** [comment]
The @plone/framework-team approved the backport of this PR to 5.2 https://community.plone.org/t/fwt-meeting-minutes-2020-05-26/12344

**Comment by mauritsvanrees** [comment]
I checked again. Started with a fresh Plone 5.2.1 site, copied its database to coredev, ran the Plone upgrade. The theming control panel looked fine, but when I save the Advanced tab, with no changes, I get this error:

```
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 162, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 359, in publish_module
  Module ZPublisher.WSGIPublisher, line 262, in publish
  Module ZPublisher.mapply, line 85, in mapply
  Module ZPublisher.WSGIPublisher, line 63, in call_object
  Module plone.app.theming.browser.controlpanel, line 65, in __call__
  Module plone.app.theming.browser.controlpanel, line 215, in update
  Module plone.registry.recordsproxy, line 59, in __setattr__
AttributeError: custom_css_timestamp
```

**Comment by mauritsvanrees** [comment]
Ah, `plone.app.theming` has its own GS profile, and `CMFPlone` automatically [applies its upgrade steps](https://github.com/plone/Products.CMFPlone/blob/5.2.1/Products/CMFPlone/MigrationTool.py#L100). So if we add an upgrade step to reload the settings, this fixes the problem. Done in https://github.com/plone/plone.app.theming/pull/184 in this commit: https://github.com/plone/plone.app.theming/pull/184/commits/a360731d9a56f506480a1146965ac1299f0d82e7.

Technically, the `plone.app.upgrade` [changes for 6.0](https://github.com/plone/plone.app.upgrade/pull/230/files) could be reverted, but it's fine to leave them (and the part of those changes that creates a `v60` structure is still needed anyway).

So that only leaves backporting the CMFPlone changes of the current PR to 5.2.

**Comment by mauritsvanrees** [comment]
I made a PR: https://github.com/plone/Products.CMFPlone/pull/3117

**Pull request merged**